### PR TITLE
fix: Remove scrollbars in edit mode

### LINKED ISSUE #20806: [Bug]: Scrollbars in container on hover of widgets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we have an auto height enabled widget (parent), with a widget (child) at the bottom most row of the widget, when we hover over the child widget, we see that the parent widget shows a scroll.

This happens in the edit mode.

https://user-images.githubusercontent.com/103687/220144477-98081b0f-0032-4e2a-9c17-1da49aaee414.mov

### Steps To Reproduce

1. Add container widget (it should have auto height enabled by default)
2. Add button widget to the bottom most row of the container widget
3. Hover over the button widget.

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud and Self hosted 1.9.8

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
- Due to the resize handle dot (see screenshot)

Screenshot 2023-02-20 at 8 39 24 PM

A scroll used to appear (before this PR) when a widget was at the bottom of an auto height enabled container like widget. (See _before_ recording below in the media section)

- This used to happen only in the edit mode, as the resize handles are not available in any other mode (view or preview)
- This PR fixes it by ensuring the parent container accommodates by removing the scrollbars when we have auto height enabled in container like widgets.
- As a result, despite the dots being present, the scrollbars don't appear in the container like widgets when auto height is enabled. (see _after_ recording below in the media section)

Fixes #20806
Media

Before

https://user-images.githubusercontent.com/103687/220143436-c4f95377-0f38-4340-ba12-c5f352bd3511.mov

After

https://user-images.githubusercontent.com/103687/220143336-bd475c52-efb8-4d70-8c21-09a7b663133a.mov

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/utils/autoHeight/mutateDOM.ts [hunk 1]
```diff
@@ -107,6 +107,7 @@ export function directlyMutateDOMNodes(
 
       if (dropTarget) {
         const dropTargetHeight =
+          4 +
```
Comment (rahulramesha) [inline app/client/src/utils/autoHeight/mutateDOM.ts:110]: NIT: Can this be stored in a constant somewhere

**Review by rahulramesha — DISMISSED** [review]

File: app/client/src/utils/autoHeight/mutateDOM.ts (hunk 1 above)
Comment (riodeuno) [reply app/client/src/utils/autoHeight/mutateDOM.ts:110]: Yes. I'll do it.

**Review by marks0351 — DISMISSED** [review]

**Comment by kamakshibhat-appsmith** [comment]
@riodeuno we are seeing scroll with Auto height with limits in all scenarios [LOOM](https://www.loom.com/share/6df256cfe62a466d91efb0386450c3f0)

File: app/client/src/widgets/TabsWidget/widget/index.tsx [hunk 2]
```diff
@@ -323,9 +324,13 @@ class TabsWidget extends BaseWidget<
       width:
         (rightColumn - leftColumn) * parentColumnSpace - WIDGET_PADDING * 2,
     };
+    const isAutoHeightEnabled: boolean =
+      isAutoHeightEnabledForWidget(this.props) &&
+      !isAutoHeightEnabledForWidget(this.props, true);
```
Comment (IAmAnubhavSaini) [inline app/client/src/widgets/TabsWidget/widget/index.tsx:329]: This is an example that proves that passing boolean values as flags is counter-intuitive. The code reads like `x = isX && !isX`, and now reviewer must go and check what is in the function `isAutoHeightEnabledForWidget`.

File: app/client/src/utils/autoHeight/mutateDOM.ts [hunk 3]
```diff
@@ -105,8 +105,14 @@ export function directlyMutateDOMNodes(
         widgetNode.style.top = `${updates[widgetId].y}px`;
       }
 
+      // In edit mode, the resize handles have this dot which is 4px in size
+      // This dot is not visible in the published mode
+      // So we need to add 4px to the drop target height to accommodate for this
```
Comment (IAmAnubhavSaini) [inline app/client/src/utils/autoHeight/mutateDOM.ts:110]: A suggestion:
These comments (starting with `//`) have a tendency to move up on auto-format to previous line, if the line above was not empty. That's one problem with it. Another idea is that multiline comments should be `/* */` ones. OR even better `/*** ... jsdoc...` ones.

File: app/client/src/widgets/TabsWidget/widget/index.tsx (hunk 2 above)
Comment (riodeuno) [reply app/client/src/widgets/TabsWidget/widget/index.tsx:329]: I agree. We already have a task to refactor this, I'll do this once we've fully shipped the auto height feature.

File: app/client/src/utils/autoHeight/mutateDOM.ts (hunk 3 above)
Comment (riodeuno) [reply app/client/src/utils/autoHeight/mutateDOM.ts]: This code has been removed. However, I've noted your point and will take care of this moving forward.

**Comment by riodeuno** [comment]
@kamakshibhat-appsmith Thanks for noticing this. I've updated the PR. Please QA now.

**Comment by kamakshibhat-appsmith** [comment]
LGTM ! @riodeuno

**Pull request merged**